controller/cache/cluster.go

```diff
@@ -405,10 +405,16 @@ func (c *clusterInfo) sync() (err error) {
 }
 
 func (c *clusterInfo) ensureSynced() error {
+	c.lock.RLock()
```

Comment:
I think we should remove this. It will re-introduce lock contention that was eliminated earlier and was not the root cause of the problem.

Reply:
done